src/common/kube-helpers.ts

```diff
@@ -16,16 +16,50 @@ function resolveTilde(filePath: string) {
   return filePath;
 }
 
+function loadFromFile(file: string) {
+  return loadFromString(fse.readFileSync(file, 'utf8'));
+}
+
+function loadFromString(config: string) {
+  try {
+    const obj = yaml.safeLoad(config);
+
+    // Validate the context structure for invalid context objects
+    if ('contexts' in obj) {
+      for (const [i, value] of obj["contexts"].entries()) {
+        if (!value.name || !value.context.cluster || !value.context.user) {
+          logger.error(`loadFromString: context ${JSON.stringify(value)} in kubeconfig has an error so will not be made available to Lens`);
+          obj["contexts"].splice(i, 1);
+        }
+      }
+    }
+    return yaml.dump(obj);
```

Comment:
what if `obj["contexts"]` is empty at this point? Should we throw here instead? What does the user see?

Reply:
So, if we do not have any contexts in the config then we will see an error in the console which will have the config supplied and the message `Error - TypeError: Cannot read property 'entries' of null`

Lens will still start up and when the user navigates to the Add Cluster screen then will see the error thrown from this block in the validateConfig function:

```
if (!config.contexts || config.contexts.length == 0) {
    throw new Error("No contexts provided in config")
}
```

<img width="514" alt="Screen Shot 2020-10-14 at 11 16 58" src="https://user-images.githubusercontent.com/14779471/95975990-cc02d200-0e0e-11eb-9955-cab339a89d7a.png">

Is this ok @jim-docker ?

Reply:
Sounds ok, since this PR doesn't really specify the UX, which can be addressed in another PR, if necessary.
It might be more accurate to say `"No valid contexts provided in config"`

Reply:
I tried this out with some bad configs and I think there needs to be more feedback to the user. If I have a config that I know has 3 contexts but only two are offered up in the "Selected contexts" drop-down, it is very puzzling. Something like
```
  if (obj["contexts"].length < config["contexts"].length) {
    notify user that "one or more contexts could not be added"
  }
```